CodeReview/src/CodeReviewManager.h

```diff
@@ -28,19 +28,41 @@
 
 #include "codereview_api.h"
 
+#include "VersionControlUI/src/nodes/DiffFrame.h"
+
 #include "nodes/CommentedNode.h"
 
 namespace CodeReview {
 
+using DiffFrames = QList<VersionControlUI::DiffFrame*>;
+
+using OrderedDiffFrames = QList<VersionControlUI::DiffFrame*>;
+
+using GroupedDiffFrames = QList<QList<VersionControlUI::DiffFrame*>>;
+
+using GroupingFunction =
+		std::function<GroupedDiffFrames (DiffFrames diffFrames)>;
+
+using OrderingFunction =
+		std::function<OrderedDiffFrames (DiffFrames diffFrames)>;
+
 class CODEREVIEW_API CodeReviewManager
 {
 	public:
 		CommentedNode* commentedNode(QString nodeId);
 		static CodeReviewManager& instance();
 
+		static QList<QList<VersionControlUI::DiffFrame*>> orderDiffFrames(
+				GroupingFunction groupingFunction, OrderingFunction orderingFunction,
+				QList<VersionControlUI::DiffFrame*> diffFrames);
+
+
 	private:
 		QHash<QString, CommentedNode*> commentedNodes_;
 		CodeReviewManager(QString oldVersion, QString newVersion);
+
+		Model::TreeManager* manager_{};
```

Comment:
We don't need this. You passed the test! 

(Should not be in there, I'll remove it)